Chunking.md

```diff
@@ -0,0 +1,84 @@
+# SaltyRTC Chunking
+
+This specification describes the binary data chunking algorithm used by
+the SaltyRTC WebRTC task. It allows the user to split up large binary
+messages into multiple chunks of a certain size.
+
+## Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Chunk Size
+
+The term *chunk size* is referring to the number of bytes of the *chunk
+header* and the *chunk data* combined.
+
+## Chunk Header
+
+When converting data to chunks, a 9 byte header MUST be prepended to
+each chunk. This allows for sending chunk over a network in any order.
+
+### Length
+
+The header is 9 bytes long.
+
+### Format
+
+The header is encoded in binary using network-oriented format (most
+significant byte first, also known as big-endian). It is structured as
+follows:
+
+    |O|IIII|SSSS|
+
+    - O: Options bit field (1 byte)
+    - I: Message id (4 bytes)
+    - S: Serial number (4 bytes)
+
+**Options bit field**
+
+The *options bit field* field is used to encode additional information
+about a chunk. Right now, only the least significant bit is being used.
+The other bits are reserved and MUST be set to `0`.
+
+    MSB           LSB
+    +---------------+
+    |0 0 0 0 0 0 0 E|
+    +---------------+
+
+    - E: End-of-message, this MUST be set to 1 if this is the
+         last chunk of the message. Otherwise, it MUST be set
+         to 0.
+
+**Message id**
+
+The *message id* SHALL be any 4 byte number. It's RECOMMENDED to start
```

Comment:
You mean because of the way the incrementing is worded? Should I write "increment the counter after each message" statt "increment the counter for each message"?